src/main/java/org/truffleruby/language/supercall/LookupSuperMethodNode.java

```diff
@@ -82,4 +90,27 @@ protected int getCacheLimit() {
         return getContext().getOptions().METHOD_LOOKUP_CACHE;
     }
 
+    private DeclarationContext getDeclarationContext(VirtualFrame frame, InternalMethod currentMethod,
```

Comment:
Should also not receive the frame because it includes HashMap operations below.
Callers can pass `RubyArguments.getDeclarationContext(frame)` instead of the frame itself.
And then this method can be @TruffleBoundary as well.